### PR TITLE
Fixing query cancel call routing

### PR DESCRIPTION
**Pull request by puneetjaiswal**
https://github.com/lyft/presto-gateway/issues/33

### CODE REVIEW

File: gateway/src/main/java/com/lyft/data/gateway/handler/QueryIdCachingProxyHandler.java [hunk 1]
```diff
@@ -141,11 +144,32 @@ public String rewriteTarget(HttpServletRequest request) {
...
   }
 
-  protected String extractQueryIdIfPresent(String path, String queryParams) {
+  protected String extractQueryIdIfPresent(HttpServletRequest request) {
+    String path = request.getRequestURI();
+    String queryParams = request.getQueryString();
+    try {
+      String queryText = CharStreams.toString(request.getReader());
+      if (!Strings.isNullOrEmpty(queryText)
+          && queryText.toLowerCase().contains("system.runtime.kill_query")) {
```
Comment (shivtools) [inline gateway/src/main/java/com/lyft/data/gateway/handler/QueryIdCachingProxyHandler.java]: Why do we want extract query id only if the string contains `system.runtime.kill_query`?

Comment (puneetjaiswal) [reply gateway/src/main/java/com/lyft/data/gateway/handler/QueryIdCachingProxyHandler.java]: Please take a look - https://github.com/lyft/presto-gateway/issues/33

Comment (shivtools) [reply gateway/src/main/java/com/lyft/data/gateway/handler/QueryIdCachingProxyHandler.java]: My concern here is that adding this condition will omit the following case for a given query: 

```
`queryText.toLowerCase().contains("system.runtime.kill_query") = false` and `!Strings.isNullOrEmpty(queryText) = true` 
```

I don't have much context here, but just want to make sure the above case is handled in the rest of the code?

Comment (shivtools) [reply gateway/src/main/java/com/lyft/data/gateway/handler/QueryIdCachingProxyHandler.java]: lgtm otherwise :)

**Pull request merged**